src/decompress.rs

```diff
@@ -494,6 +541,43 @@ impl<'src> DecompressStarted<'src> {
         Some(image_dst)
     }
 
+    /// Supports any pixel type that is marked as "plain old data", see bytemuck crate.
+    /// `[u8; 3]` and `rgb::RGB8` are fine, for example.
+    /// Allocation-less version of `read_scanlines`
+    #[track_caller]
+    pub fn read_scanlines_into<T: rgb::Pod>(&mut self, dest: &mut [T]) -> usize {
```

Comment:
Maybe you'd like to read into `[MaybeUninit<T>]`?

Reply:
I'm absolutely unfamiliar with `MaybeUninit`, so I have a small question, API-wise; Can a `&mut [u8]` be transparently coreced into a `&mut [MaybeUninit<T>]`?

Also AFAIK it would require the consumer to then `std::mem::transmute` to get back their concrete `&mut [u8]`, which seems counter-intuitive to me (having to use `unsafe` calling a safe wrapper)

Another point for `&mut [u8]` is that it mirrors `read_scanlines`'s API.

Reply:
It does need transmute for now. There are unstable APIs in libstd that will make it unnecessary in the future.